File: historyscan-back/src/main/java/io/sch/historyscan/domain/contexts/analysis/common/EnumAnalysisType.java
package io.sch.historyscan.domain.contexts.analysis.common;

import java.util.Arrays;
import java.util.Optional;

public enum EnumAnalysisType {
    COMMITS_SCAN("history"),
    CLOC_REVISIONS("cloc-revisions"),
    NETWORK("network");
    private final String title;

    EnumAnalysisType(String title) {
        this.title = title;
    }

    public String getTitle() {
        return title;
    }

    public static Optional<EnumAnalysisType> fromTitle(String text) {
        return Arrays.stream(EnumAnalysisType.values())
                .filter(b -> b.title.equalsIgnoreCase(text))
                .findFirst();
    }
}

File: historyscan-back/src/main/kotlin/io/sch/historyscan/domain/contexts/analysis/common/EnumAnalysisType.kt
package io.sch.historyscan.domain.contexts.analysis.common

import java.util.*

enum class EnumAnalysisType(@JvmField val title: String) {
    COMMITS_SCAN("history"),
    CLOC_REVISIONS("cloc-revisions"),
    NETWORK("network");

    companion object {
        fun fromTitle(text: String?): Optional<EnumAnalysisType> {
            return Arrays.stream(entries.toTypedArray())
                    .filter { b: EnumAnalysisType -> b.title.equals(text, ignoreCase = true) }
                    .findFirst()
        }
    }
}
